### ISSUE TITLE
Compiler warning regarding ToDictionary method

### ISSUE THREAD

**Issue by vesnx** [Cysharp/MemoryPack#177]
I get a compiler warning in MemoryPack.Generator that perhaps needs consideration
The class:
MemoryPackGenerator.Emitter.cs

The code:
   ```
 string EmitConstructor()
    {
        // no need `;` because after using object initializer
        if (this.Constructor == null || this.Constructor.Parameters.Length == 0)
        {
            return $"new {TypeName}()";
        }
        else
        {
            var nameDict = Members.Where(x => x.Symbol != null && x.IsConstructorParameter).ToDictionary(x => x.ConstructorParameterName, x => x.Name, StringComparer.OrdinalIgnoreCase);//--> line 899
            var parameters = this.Constructor.Parameters
                .Select(x =>
                {
                    if (nameDict.TryGetValue(x.Name, out var name))
                    {
                        return $"__{name}";
                    }
                    return null; // invalid, validated.
                })
                .Where(x => x != null);

            return $"new {TypeName}({string.Join(", ", parameters)})";
        }
    }
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS8714	The type 'string?' cannot be used as type parameter 'TKey' in the generic type or method 'Enumerable.ToDictionary<TSource, TKey, TElement>(IEnumerable<TSource>, Func<TSource, TKey>, Func<TSource, TElement>, IEqualityComparer<TKey>?)'. Nullability of type argument 'string?' doesn't match 'notnull' constraint.	MemoryPack.Generator	..\src\MemoryPack.Generator\MemoryPackGenerator.Emitter.cs	899	Active

**Comment by olivier-spinelli** [comment]
Trying to help here:
- MS has nothing to do with this.
- In `... .ToDictionary(x => x.ConstructorParameterName, ...`, x.ConstructorParameterName is a nullable string.
=> To suppresse the warning, `... .ToDictionary(x => x.ConstructorParameterName!, ...` (adding the bang! operator) should be enough.

**Comment by vesnx** [comment]
sure, however it's not my code

**Comment by olivier-spinelli** [comment]
Of course. You may create a PR for this.

**Comment by hadashiA** [comment]
Fixed in #232.
